packages/account/test/fuel-core-schema.test.ts

```diff
@@ -21,6 +23,21 @@ function generateChecksum(
  * @group node
  */
 describe('fuel-core-schema.graphql', () => {
+  let destroy: () => void;
+
+  beforeEach(async () => {
+    const { cleanup } = await setupTestProviderAndWallets({
+      nodeOptions: {
+        port: '4000',
+      },
+    });
+    destroy = cleanup;
+  });
+
+  afterEach(() => {
+    destroy();
+  });
+
```

Comment:
I suggest you change the `build:schema` script to run a node with `launchNode` and get the schema from it:
```ts
// get-fuel-core-schema.mts
const { url, cleanup } = await launchNode();

execSync(`get-graphql-schema ${url} > src/providers/fuel-core-schema.graphql && prettier --write src/providers/fuel-core-schema.graphql`);

cleanup();
```
Then you can remove these `beforeEach`/`afterEach` blocks.

Reply:
I actually prefer this current setup as putting a lengthy command in an `execSync` is less readable and less standard in a test set up as opposed to the test hooks designed for this purpose. Also both setups accomplish the same purpose of setup/teardown.

Reply:
My comment wasn't related to the test per se but to the `build:schema` script. The main point is in making the `build:schema` command self-sufficient. As it is currently, it's expecting a node to be already running on `http://127.0.0.1:4000/v1/graphql`, hence why you had to do the setup/teardown. 

Once the logic is placed into `get-fuel-core-schema.mts` you can change the `build:schema` command to be:
```json
"build:schema": "tsx path/to/get-fuel-core-schema.mts"
```

and then you will be able to remove this test's setup/teardown because the script is running a node for itself instead of expecting one on a specific port.

Looking at it from a broader perspective, one could say that this is part of the same endeavor to _not_ require a node to run on port 4000.

Reply:
Thanks for the clarification @nedsalk I agree with you now. 

I was trying to do this locally with a `launchTestNode` script but I was running into a strange issue

```
Error: an error occurred while loading the chain state file: Some("/var/folders/fl/djllw9k50cqcttx_fmrx0fsc0000gn/T/.fuels/a56debc7-014b-491f-a2dc-810f378eeda6/snapshotDir/chainConfig.json")

Caused by:

_FuelError: Error: an error occurred while loading the chain state file: Some("/var/folders/fl/djllw9k50cqcttx_fmrx0fsc0000gn/T/.fuels/a56debc7-014b-491f-a2dc-810f378eeda6/snapshotDir/chainConfig.json")
```

I think we can do this in a follow-up since it's not worth going down that rabbit hole for the purposes of this PR.

https://github.com/FuelLabs/fuels-ts/issues/2938